analyzer/runtime/abiparse/abiparse.go

```diff
@@ -39,7 +39,7 @@ func evmPreMarshal(v interface{}, t abi.Type) interface{} {
 	return v
 }
 
-func EVMParseData(data []byte, contractABI *abi.ABI) (*abi.Method, []interface{}, error) {
+func ParseData(data []byte, contractABI *abi.ABI) (*abi.Method, map[string]interface{}, error) {
```

Comment:
ya, the order will live on in the ABI itself

Reply:
I'm thinking about how we'll serialize these into the DB (and later present them in the API).
The `abi.Method` does preserve the order, as you say. But the DB (and the API field `method_name` or whatever) will likely be just a string. So in the DB, I expect we'll want the args as a list (or ordered dict, but that's not a JSON type). If there are analyzer-internal benefits to having args as a dict then we can do the conversion to a list when writing to the DB. Otherwise, probably best/simplest to generate a list here already.

Reply:
alright good, I've locally switched back to unpacking into a list. I'm working on getting the events parsed as a list too

Reply:
adding doc comment, switched back to arg list